plugins/org.jboss.tools.openshift.io.core/src/main/java/org/jboss/tools/openshift/io/core/AccountService.java

```diff
@@ -0,0 +1,162 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.io.core;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.ParseException;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.eclipse.core.resources.IResource;
+import org.jboss.tools.openshift.io.core.exception.OpenshiftIOConfigurationException;
+import org.jboss.tools.openshift.io.core.exception.OpenshiftIOLoginException;
+import org.jboss.tools.openshift.io.core.exception.OpenshiftIORefreshException;
+import org.jboss.tools.openshift.io.core.model.ICluster;
+import org.jboss.tools.openshift.io.core.model.IAccount;
+import org.jboss.tools.openshift.io.core.model.IAccountModel;
+import org.jboss.tools.openshift.io.internal.core.model.AccountModel;
+
+public class AccountService {
+
+	private static final AccountService INSTANCE = new AccountService();
+
+	private final LoginProvider provider = LoginProvider.get();
+
+	private HttpClient client = HttpClients.createDefault();
+
+	private IAccountModel model;
+
+	private AccountService() {
+	}
+
+	public static AccountService getDefault() {
+		return INSTANCE;
+	}
+
+	public IAccountModel getModel() {
+		if (null == model) {
+			model = new AccountModel();
+		}
+		return model;
+	}
+
+	public AccountStatus getStatus(IAccount account) {
+		if (account.getAccessToken() == null) {
+			return AccountStatus.NEEDS_LOGIN;
+		}
+		long lastRefreshed = account.getLastRefreshedTime();
+		long current = System.currentTimeMillis();
+		if (current > account.getAccessTokenExpiryTime()) {
+			if (current > account.getRefreshTokenExpiryTime()) {
+				return AccountStatus.NEEDS_LOGIN;
+			} else {
+				return AccountStatus.NEEDS_REFRESH;
+			}
+		}
+		if (((current - lastRefreshed) > OSIOCoreConstants.DURATION_24_HOURS)
```

Comment:
might improve readability if this was extracted to #wasRefreshed24hAgo

Reply:
+1